sdk/test-utils/perf/src/perfTestBase.ts

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import minimist, { ParsedArgs as MinimistParsedArgs } from "minimist";
+import {
+  PerfOptionDictionary,
+  parsePerfOption,
+  DefaultPerfOptions,
+  defaultPerfOptions,
+  validateOptions,
+  ParsedPerfOptions
+} from "./options";
+import { PerfParallel } from "./parallel";
+import { AbortController } from "@azure/abort-controller";
+
+/**
+ * Defines the behavior of the PerfTest constructor, to use the class as a value.
+ */
+export interface PerfTestConstructor<
+  TOptions extends Record<string, unknown> = Record<string, unknown>
+> {
+  new (): PerfTestBase<TOptions>;
+}
+
+/**
+ * Conveys the structure of any Perf test.
+ * It allows developers to define the optional parameters to receive
+ * in a way that aids them to auto-complete these options while adding code inside of each method.
+ * It provides methods to parse the assigned options, as well as to set up and tear down any resources
+ * at a global level (only one setup and teardown for the whole execution of a Perf test),
+ * and at a local level, which happens once for each initialization of the test class
+ * (initializations are as many as the "parallel" command line parameter specifies).
+ */
+export abstract class PerfTestBase<TOptions = Record<string, unknown>> {
+  public abstract options: PerfOptionDictionary<TOptions>;
+
+  public get parsedOptions(): ParsedPerfOptions<TOptions & DefaultPerfOptions> {
+    // Only validate the options if they are defined: if (when) parsedOptions is called
+    // in the constructor, options will be undefined.
+    if (this.options) {
+      validateOptions({
+        ...this.options,
+        ...defaultPerfOptions
+      });
+    }
+
+    // This cast is needed because TS thinks
+    //   PerfOptionDictionary<TOptions & DefaultPerfOptions>
+    //   is different from
+    //   PerfOptionDictionary<TOptions> & PerfOptionDictionary<DefaultPerfOptions>
```

Comment:
```suggestion
    // This cast is needed because TS thinks that:
    // ```ts
    // PerfOptionDictionary<TOptions & DefaultPerfOptions>
    // ```
    // Is different from:
    // ```ts
    // PerfOptionDictionary<TOptions> & PerfOptionDictionary<DefaultPerfOptions>
    // ```
```

Proper markdown 🙂

Reply:
it's a comment block, where would it be anywhere used?

Reply:
Well, in case this gets ever rendered by some doc parser, we would benefit from seeing the code properly colored etc ☺️